### PR TITLE
tracee-ebpf: extract logic into pkg/cmd

### PR DESCRIPTION
**Pull request by josedonizetti**
Fix https://github.com/aquasecurity/tracee/issues/2415

This is a refactoring, nothing should have changed besides restructuring. 

This PR extracts the `cmd/tracee-ebpf` login into a new package `pkg/cmd`. This is necessary because we will be using this logic in multiple binaries (perhaps even inside aqua apps). It also breaks the booting into two steps:

1 - parsing flags and returning configurations
2 - running tracee

This is done because we want to decouple the flags library from running tracee, allowing us to change the library in the future, but also others using tracee as a lib. 

For extra context check the PR https://github.com/aquasecurity/tracee/pull/2374

### CODE REVIEW

File: cmd/tracee-rules/main.go [hunk 1]
```diff
@@ -136,26 +136,18 @@ func main() {
...
+			)
 
-				if c.Bool(server.PProfEndpointFlag) {
-					httpServer.EnablePProfEndpoint()
-				}
+			if err != nil {
+				return err
+			}
 
+			if httpServer != nil {
```
Comment (geyslan) [inline cmd/tracee-rules/main.go:150]: Should we have a log if httpServer is nil?

Comment (josedonizetti) [reply cmd/tracee-rules/main.go:150]: hmm...why? if the user didn't pass any flag to enable it, why log?

Comment (geyslan) [reply cmd/tracee-rules/main.go:151]: I got it. I was already rethinking my question, since rules are going to die.

**Review by geyslan — APPROVED** [review]
LGTM

**Review by rafaeldtinoco — APPROVED** [review]
LGTM as well.

**Pull request merged**